tests/test_interface.rb

```diff
@@ -165,6 +170,37 @@ def interface_supports_property?(intf, message)
     flunk(message)
   end
 
+  # Helper to find first valid speed that isn't "auto"
+  def valid_speed_set(interface)
+    valid_speed = nil
+    speeds = capable_speed_values(interface)
+    speeds.each do |value|
+      next if value == 'auto'
+      begin
+        interface.speed = value
+        assert_equal(value.to_i, interface.speed)
+      rescue Cisco::CliError => e
+        next if speed_change_disallowed?(e.message)
+        raise
```

Comment:
I prefer raising an error than reporting the test as a "failure" when the CLI errors after a config command.  It's usually an indicator to me that our YAML needs to be fixed, either by `exclude` or by specifying a different `default_value`